### ISSUE TITLE
Use rustyline crate for command line input

### ISSUE THREAD

**Issue by sphinxc0re** [DeMille/encrusted#1]
The [rustyline](https://crates.io/crates/rustyline/) crate provides a nice and clean way to set up an interactive console while supplying tools for creating a history and navigating the command line.

**Comment by Lokathor** [comment]
That project is [quite possibly dead](https://github.com/kkawakam/rustyline/issues/123), even though it is 1.0

The [linefeed](https://github.com/murarth/linefeed) project is similar, but still active (but not 1.0).

**Comment by sphinxc0re** [comment]
But it works perfectly and is totally stable. So why should we use linefeed instead? Rustyline has a very nice API.

**Comment by Lokathor** [comment]
Well _any_ version of any crate is stable if you just never switch versions :P so I'm not sure how "1.0 and it will never update" is in any way better than "0.5 but it will get updates". At least in the latter case there's a chance that newer versions will break little and give you some nice new ability in exchange.

But particularly to your question of what linefeed gets right: linefeed allows for concurrent output to happen while a line-reading is taking place without disrupting the line-reading prompt and partial input from being displayed properly. As far as I can tell rustyline has no similar ability.

**Comment by sphinxc0re** [comment]
I mean, as long as it cleans up the UX of the command prompt it's fine.

**Comment by DeMille** [comment]
Yeah I definitely want to take a look into these. Right now the terminal UI is suuuper basic. I struggled for a while to make sure I could build both the wasm and normal targets, but I think I have them separated enough now to introduce something like this.
